content/php/concepts/arrays/terms/extract/extract.md

```diff
@@ -0,0 +1,77 @@
+---
+Title: 'extract()'
+Description: 'Uses array keys as variable names and values as variable values and for each element creates a variable in the current symbol table'
+Subjects:
+  - 'Computer Science'
+  - 'Web Design'
+  - 'Web Development'
+Tags:
+  - 'Arrays'
+  - 'Functions'
+CatalogContent:
+  - 'learn-php'
+  - 'paths/computer-science'
+---
+
+The **`extract()`** method imports variables into the current symbol table from an array.
+
+## Syntax
+
+```pseudo
+extract($array, flags, prefix)
+```
+
+The `extract()` function has one required parameter and some optional parameters:
+
+- `$array`: Specifies the input `array`.
+- `flags`: Optional.The extract() function checks for invalid variable names and collisions with existing variable names. This parameter specifies how invalid and colliding names are treated.
+Possible values:
+    - EXTR_OVERWRITE - Default. On collision, the existing variable is overwritten
+    - EXTR_SKIP - On collision, the existing variable is not overwritten
+    - EXTR_PREFIX_SAME - On collision, the variable name will be given a prefix
+    - EXTR_PREFIX_ALL - All variable names will be given a prefix
+    - EXTR_PREFIX_INVALID - Only invalid or numeric variable names will be given a prefix
+    - EXTR_IF_EXISTS - Only overwrites existing variables in the current symbol table, otherwise does nothing
+    - EXTR_PREFIX_IF_EXISTS - Only adds prefix to variables if the same variable exists in the current symbol table
+    - EXTR_REFS - Extracts variables as references.
```

Comment:
```suggestion
    - `EXTR_OVERWRITE`: Default. On collision, the existing variable is overwritten
    - `EXTR_SKIP`: On collision, the existing variable is not overwritten
    - `EXTR_PREFIX_SAME`: On collision, the variable name will be given a prefix
    - `EXTR_PREFIX_ALL`: All variable names will be given a prefix
    - `EXTR_PREFIX_INVALID`: Only invalid or numeric variable names will be given a prefix
    - `EXTR_IF_EXISTS`: Only overwrites existing variables in the current symbol table, otherwise does nothing
    - `EXTR_PREFIX_IF_EXISTS`: Only adds prefix to variables if the same variable exists in the current symbol table
    - `EXTR_REFS`: Extracts variables as references.
```